drivers/ethernet/eth_stm32_hal.c

```diff
@@ -957,10 +957,61 @@ void HAL_ETH_TxCpltCallback(ETH_HandleTypeDef *heth_handle)
 #if defined(CONFIG_ETH_STM32_HAL_API_V2)
 void HAL_ETH_ErrorCallback(ETH_HandleTypeDef *heth)
 {
-	/* Do nothing */
-	/* Do not log errors. If errors are reported du to high traffic,
+	/* Do not log errors. If errors are reported due to high traffic,
 	 * logging errors will only increase traffic issues
 	 */
+
+	uint32_t dma_error, mac_error;
```

Comment:
I propose to put this whole section under `IF_ENABLED(CONFIG_NET_STATISTICS_ETHERNET)`.
This will save cpu cycles and memory when not enabled.

Reply:
Thanks @erwango, I've conditionally compiled the code based on the config option.